modules/openapi-generator-cli/src/main/java/org/openapitools/codegen/cmd/GenerateBatch.java

```diff
@@ -0,0 +1,261 @@
+import com.fasterxml.jackson.databind.deser.std.DelegatingDeserializer;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import io.airlift.airline.Arguments;
+import io.airlift.airline.Command;
+import io.airlift.airline.Option;
+import org.openapitools.codegen.ClientOptInput;
+import org.openapitools.codegen.CodegenConfig;
+import org.openapitools.codegen.DefaultGenerator;
+import org.openapitools.codegen.config.CodegenConfigurator;
+import org.openapitools.codegen.config.DynamicSettings;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Locale;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@SuppressWarnings({"unused", "MismatchedQueryAndUpdateOfCollection"})
+@Command(name = "batch", description = "Generate code in batch via external configs.", hidden = true)
+public class GenerateBatch implements Runnable {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(GenerateBatch.class);
+
+    @Option(name = {"-v", "--verbose"}, description = "verbose mode")
+    private Boolean verbose;
+
+    @Option(name = {"-r", "--threads"}, description = "thread count")
+    private Integer threads;
+
+    @Arguments(description = "Generator configuration files.", required = true)
+    private List<String> configs;
+
+    @Option(name = {"--fail-fast"}, description = "fail fast on any errors")
+    private Boolean failFast;
+
+    @Option(name = {"--timeout"}, description = "execution timeout (minutes)")
+    private Integer timeout;
+
+    @Option(name = {"--includes-base-dir"}, description = "base directory used for includes")
+    private String includes;
+
+    @Option(name = {"--root-dir"}, description = "root directory used output/includes (includes can be overridden)")
+    private String root;
+
+    /**
+     * When an object implementing interface <code>Runnable</code> is used
+     * to create a thread, starting the thread causes the object's
+     * <code>run</code> method to be called in that separately executing
+     * thread.
+     * <p>
+     * The general contract of the method <code>run</code> is that it may
+     * take any action whatsoever.
+     *
+     * @see Thread#run()
+     */
+    @Override
+    public void run() {
+        if (configs.size() < 1) {
+            LOGGER.error("No configuration file inputs specified");
+            System.exit(1);
+        }
+
+        int cores = Runtime.getRuntime().availableProcessors();
+        int numThreads = 2 * cores;
+        if (null != threads && (threads > 0 && threads < Thread.activeCount())) {
+            numThreads = threads;
+        }
+
```

Comment:
Minor suggestion: maybe good to use LOGGER.info to show how many threads are used as an FYI to the user.